### PR TITLE
[VEN-1911]: PeckShield audit fix for token converter

### PR DESCRIPTION
**Pull request by Debugger022**
## Description

Resolves # VEN-1911

### CODE REVIEW

File: contracts/ProtocolReserve/RiskFundStorage.sol [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
     bytes32 private __deprecatedSlot1;
-    bytes32 private __deprecatedSlot2;
-    bytes32 private __deprecatedSlot3;
 
     /// @notice Available asset's fund per pool in RiskFund
     /// Comptroller(pool) -> Asset -> amount
     mapping(address => mapping(address => uint256)) public poolAssetsFunds;
 
+    bytes32 private __deprecatedSlot2;
+    bytes32 private __deprecatedSlot3;
```
[the remark this answers is not in the document]
Comment (Debugger022) [reply contracts/ProtocolReserve/RiskFundStorage.sol:18]: Added [5dd0548](https://github.com/VenusProtocol/protocol-reserve/pull/17/commits/5dd0548efd835485d60e40d1248c59ed1a876d49)

File: contracts/ProtocolReserve/RiskFundStorage.sol [hunk 2]
```diff
@@ -30,22 +34,19 @@ contract MaxLoopsLimitHelpersStorage {
 /// @author Venus
 /// @dev Risk fund V1 storage
 contract RiskFundV1Storage is ReserveHelpersStorage, MaxLoopsLimitHelpersStorage {
+    address public convertibleBaseAsset;
+    address public shortfall;
```
[the remark this answers is not in the document]
Comment (Debugger022) [reply contracts/ProtocolReserve/RiskFundStorage.sol:43]: Added [5dd0548](https://github.com/VenusProtocol/protocol-reserve/pull/17/commits/5dd0548efd835485d60e40d1248c59ed1a876d49)

File: contracts/TokenConverter/RiskFundConverter.sol [hunk 3]
```diff
@@ -208,16 +208,16 @@ contract RiskFundConverter is AbstractTokenConverter {
         uint256 balance = IERC20Upgradeable(tokenAddress).balanceOf(address(this));
         uint256 balanceDiff = balance - assetsReserves[tokenAddress];
 
-        uint256 amountDiff = amount - balanceDiff;
+        int256 amountDiff = int256(amount) - int256(balanceDiff);
```
Comment (chechu) [inline contracts/TokenConverter/RiskFundConverter.sol:211]: I think we could avoid the castings this way:

```
        if (balanceDiff > amount) {
            uint256 amountDiff = amount - balanceDiff;
            address[] memory pools = getPools(tokenAddress);
            uint256 assetReserve = assetsReserves[tokenAddress];
            for (uint256 i; i < pools.length; ++i) {
                uint256 poolShare = (poolsAssetsReserves[pools[i]][tokenAddress] * EXP_SCALE) / assetReserve;
                if (poolShare == 0) continue;
                updatePoolAssetsReserve(pools[i], tokenAddress, amountDiff, poolShare);
            }
            assetsReserves[tokenAddress] -= amountDiff;
        }
```

Comment (Debugger022) [reply contracts/TokenConverter/RiskFundConverter.sol]: Refactored, https://github.com/VenusProtocol/protocol-reserve/pull/17/commits/fd201247db8c1e4ee6b6dcbdf3ee37ff05f0b23e

**Review by chechu — APPROVED** [review]

**Pull request merged**